data/finset.lean

```diff
@@ -1043,6 +1043,54 @@ by haveI := classical.prop_decidable;
 
 end fold
 
+/- max and min of finite sets -/
+section max_min
+variables [decidable_linear_order α]
+
+protected def max : finset α → option α :=
+fold (option.lift_or_get max) none some
+
+@[simp] theorem max_empty : (∅ : finset α).max = none :=
+by simp [finset.max]
+
+@[simp] theorem max_insert {a : α} {s : finset α} (h : a ∉ s) :
+  (insert a s).max = option.lift_or_get max (some a) s.max :=
+by simp [finset.max, fold_insert h]
+
+@[simp] theorem max_singleton {a : α} : finset.max {a} = some a :=
+by simp [finset.max, option.lift_or_get]
+
+theorem le_max_of_mem {s : finset α} : ∀ {a : α}, a ∈ s → ∃ b, a ≤ b :=
+finset.induction_on s (by simp) $
```

Comment:
This doesn't mention `max` at all, and is a quiet boring statement: you are missing `b ∈ s`. Similar with the later statement for `min`.